packages/opentelemetry-core/src/common/attributes.ts

```diff
@@ -94,3 +122,34 @@ function isValidPrimitiveAttributeValue(val: unknown): boolean {
 
   return false;
 }
+
+function sanitizeHttpUrl(val: AttributeValue): AttributeValue | undefined {
+  let out: AttributeValue | undefined;
+
+  if (typeof val !== 'string') {
+    diag.warn(
```

Comment:
I'd find these diag warnings in the value sanitizer unexpected. I think it should be still valid for trace-sdk users to set arbitrary valid attribute values as `http.url`. `http.url` is part of the semantic conventions, not part of the SDK specifications.

@open-telemetry/javascript-approvers  WDYT?

Reply:
according to #2000, its part of the specification. i'm open to all of your guidance on how to approach this

Reply:
Yes, it's part of the semantic convention specification, not the trace sdk specification. So IMHO the constraints should only be applied to instrumentations that collect those attributes.